File: work/P634.py
from math import sqrt
def solve(n):
    primes = [1]*(int(n**(1/3))+1)
    sqfree = [1]*(int(n**(1/3))+1)
    for i in range(2, int(sqrt(len(primes))) + 1):
        if primes[i]:
            for k in range(i+i, len(primes), i):
                primes[k] = 0
            for k in range(i**2, len(primes), i**2):
                sqfree[k] = 0
    print(sqfree)
    rv = 0
    for i in range(2, len(sqfree)):
        rv += sqfree[i]*int(sqrt(n//(i*i*i)))
    return rv
print(solve(3*10**6))

File: work/P634.cpp
#include <bits/stdc++.h>
using namespace std;

#define ll long long
#define len sizeof

ll isqrt(ll n){
    if (n == 1){
        return n;
    }
    ll a = n >> 1;
    ll b = n;
    ll c;
    while (a < b){
        c = (a + n/a) >> 1;
        a,b = c,a;
    }   
    return b;
}

ll solve(ll n){
    ll rv = 0;
    bool sq[(int) pow(n, 0.3333334)];
    for (ll i = 2; i < len(sq); i++){
        for (ll k = i*i; k < len(sq); k++){
            sq[k] = false;
        }
    }
    for (int b = 2; b < len(sq); b++){
        if (sq[b] = 0){
            continue;
        }
        rv += (int) sqrt(n/b/b/b) - 1;
    }
    return rv;
}

int main(){
    cout << solve(2*pow(10, 4));
}